test/differentials/composite.jl

```diff
@@ -177,8 +177,8 @@ end
 
         @testset "Internals don't allocate a ton" begin
             bk = (; x=1.0, y=2.0)
-            @test (@allocated(ChainRulesCore.construct(Foo, bk))) <= 32
-            @test (@allocated ChainRulesCore.elementwise_add(bk, bk)) <= 48
+            @test (@ballocated(ChainRulesCore.construct($Foo, $bk))) <= 32
```

Comment:
test failure is 
```julia
Internals don't allocate a ton: Test Failed at /home/travis/build/JuliaDiff/ChainRulesCore.jl/test/differentials/composite.jl:180

  Expression: #= /home/travis/build/JuliaDiff/ChainRulesCore.jl/test/differentials/composite.jl:180 =# @ballocated(ChainRulesCore.construct($Foo, $bk)) <= 32

   Evaluated: 48 <= 32
```

Maybe @oxinabox can see how this would have changed, or if we're doing something wrong here, but i suspect the answer is tp just bump this number

```suggestion
            @test (@ballocated(ChainRulesCore.construct($Foo, $bk))) <= 48
```

Reply:
It should be 32
```
julia> struct Foo
           x
           y::Float64
       end

julia> @btime Foo($(rand(1:10)), $(rand()))
  8.094 ns (1 allocation: 32 bytes)
Foo(8, 0.22091938267065747)
```

Reply:
Look its fine (on julia 1.4)
```
julia> bk
(x = 1, y = 0.5)

julia> @btime Foo($(bk)...)
  7.639 ns (1 allocation: 32 bytes)
Foo(1, 0.5)

julia> ChainRulesCore.construct(Foo, bk)
Foo(1, 0.5)

julia> @btime ChainRulesCore.construct($Foo, $bk)
  7.502 ns (1 allocation: 32 bytes)
```

Reply:
Am I good to merge despite this not passing CI?

Reply:
No. do not do that.
When something is approved that does not mean it is ok to merge.
It means it is ok to merge without further review.
You still need to fix things that are obviously wrong.

Don't merge things that break CI.

For this case it might be OK to do:

```suggestion
    @test_broken (@ballocated(ChainRulesCore.construct($Foo, $bk))) <= 32
    # weaker version of the above (which should pass, but make sure not failing too bad)
    @test (@ballocated(ChainRulesCore.construct($Foo, $bk))) <= 48
```

Reply:
> No. do not do that.

Haha thanks for the clarification. I figured probably not. I'll commit your suggestion and merge once the tests pass.